src/routes/api/dept/name/+server.ts

```diff
@@ -0,0 +1,28 @@
+import { editDeptName, getUserFromSession } from '$lib/server/database';
+import type { RequestHandler } from './$types';
+import { StatusCodes } from 'http-status-codes';
+import { error } from '@sveltejs/kit';
+
+// eslint-disable-next-line func-style
+export const PATCH: RequestHandler = async ({ cookies, request }) => {
+    const form = await request.formData();
+
+    const id = form.get('id');
+    if (id === null || id instanceof File) throw error(StatusCodes.BAD_REQUEST);
+    const did = parseInt(id, 10);
+
+    const name = form.get('name');
+    if (name === null || name instanceof File) throw error(StatusCodes.BAD_REQUEST);
+
+    const sid = cookies.get('sid');
+    if (!sid) throw error(StatusCodes.UNAUTHORIZED);
+
+    // TODO: session has expired so we must inform the client that they should log in again
+    const user = await getUserFromSession(sid);
+    if (user === null) throw error(StatusCodes.UNAUTHORIZED);
+    if (!user.admin) throw error(StatusCodes.FORBIDDEN);
```

Comment:
So here's the tricky part about this: should we let department heads edit their department names? If so, we need a dedicated database helper function that retrieves the `BOOLEAN` on whether the current agent is `head`.

Reply:
Sounds like something they should be able to do, but we can'timplement yet since the dept-agents table and the adding of agents functionality hasn't been realized yet. We can probably included that when we handle that case na.

Reply:
Alright, I'll do the helper function after merging this, then retroactively apply the permission check.